### PR TITLE
[ja] Update OpenID Connect Flow diagram

### PR DESCRIPTION
**Pull request by ystkfujii**
Fixes : #28270
Updated ja/docs/reference/access-authn-authz/authentication.md to follow v1.19 of the original (English) text.

### CODE REVIEW

**Comment by kbhawkey** [comment]
/retitle [ja] Update OpenID Connect Flow diagram

**Comment by atoato88** [comment]
Is this PR for resolving [#28270](https://github.com/kubernetes/website/issues/28270)?
If so, please update description for this PR like [this](https://github.com/kubernetes/website/pull/28181#issue-657763019) or [this](https://github.com/kubernetes/website/pull/28371#issue-667861413)?

We should link this PR to the related issue at least for smooth reviewing.

**Comment by ystkfujii** [comment]
> Is this PR for resolving [#28270](https://github.com/kubernetes/website/issues/28270)?
> If so, please update description for this PR like [this](https://github.com/kubernetes/website/pull/28181#issue-657763019) or [this](https://github.com/kubernetes/website/pull/28371#issue-667861413)?
> 
> We should link this PR to the related issue at least for smooth reviewing.

Thanks for your comment.
The description has been updated.

File: content/ja/docs/reference/access-authn-authz/authentication.md [hunk 1]
```diff
@@ -341,65 +369,154 @@ current-context: webhook
...
-    user: name-of-api-sever
+    user: name-of-api-server
   name: webhook
 ```
 
-クライアントが[上記](#putting-a-bearer-token-in-a-request)のようにBearerトークンを使用してAPIサーバーとの認証を試みた場合、認証Webhookはトークンを含むJSONでシリアライズされた`authentication.k8s.io/v1beta1` `TokenReview`オブジェクトをリモートサービスにPOSTします。Kubernetesはそのようなヘッダーが不足しているリクエストを作成しようとはしません。
+クライアントが[上記](#putting-a-bearer-token-in-a-request)のようにBearerトークンを使用してAPIサーバーとの認証を試みた場合、認証Webhookはトークンを含むJSONでシリアライズされた`TokenReview`オブジェクトをリモートサービスにPOSTします。
 
-Webhook APIオブジェクトは、他のKubernetes APIオブジェクトと同じように、[Versioning Compatibility Rule](/docs/concepts/overview/kubernetes-api/)に従うことに注意してください。実装者は、ベータオブジェクトで保証される互換性が緩いことに注意し、正しいデシリアライゼーションが使用されるようにリクエストの"apiVersion"フィールドを確認する必要があります。さらにAPIサーバーは、API拡張グループ`authentication.k8s.io/v1beta1`を有効にしなければなりません(`--runtime config=authentication.k8s.io/v1beta1=true`)。
+Webhook APIオブジェクトは、他のKubernetes APIオブジェクトと同じように、[Versioning Compatibility Rule](/docs/concepts/overview/kubernetes-api/)に従うことに注意してください。実装者は、正しいデシリアライゼーションが使用されるようにリクエストの"apiVersion"フィールドを確認する必要があり、リクエストと同じバージョンの`TokenReview`オブジェクトで応答する必要があります。
```
Comment (atoato88) [inline content/ja/docs/reference/access-authn-authz/authentication.md:378]: [In original](https://gist.github.com/inductor/18a9278caa84436a6ee82cc80f12c38f#file-en-diff-L142), a word of `must` is surrounded with `**`, you should specify with same manner.

File: content/ja/docs/reference/access-authn-authz/authentication.md [hunk 2]
```diff
@@ -341,65 +369,154 @@ current-context: webhook
...
+{{< /tabs >}}
 
 リクエストに失敗した場合は、以下のように返されます。
 
-```json
+{{< tabs name="TokenReview_response_error" >}}
+{{% tab name="authentication.k8s.io/v1" %}}
+```yaml
 {
   "apiVersion": "authentication.k8s.io/v1beta1",
```
Comment (atoato88) [inline content/ja/docs/reference/access-authn-authz/authentication.md:507]:
```suggestion
  "apiVersion": "authentication.k8s.io/v1",
```

File: content/ja/docs/reference/access-authn-authz/authentication.md [hunk 3]
```diff
@@ -341,65 +369,154 @@ current-context: webhook
...
   "kind": "TokenReview",
   "status": {
-    "authenticated": false
+    "authenticated": false,
+    # 認証に失敗した詳細をオプションで記載します。
+    # エラーが提供されていない場合、APIは一般的なUnauthorizedメッセージを返却します。
+    # authenticated=trueの場合、エラーフィールドは無視されます。
+    "error": "Credentials are expired"
   }
```
Comment (atoato88) [inline content/ja/docs/reference/access-authn-authz/authentication.md:515]: You need to add sections about [here](https://gist.github.com/inductor/18a9278caa84436a6ee82cc80f12c38f#file-en-diff-L296-L309)?

File: content/ja/docs/reference/access-authn-authz/authentication.md [hunk 4]
```diff
@@ -195,7 +195,35 @@ Secretは常にbase64でエンコードされるため、これらの値もbase6
...
+    activate user
+    user ->> kube: 3. Kubectlを使用する場合、<br>--tokenフラグでid_tokenを使用するか、<br>または.kube/configにトークンを追加する
+    deactivate user
+    activate kube
+    kube ->> api: 4. 認可: Bearer...
+    deactivate kube
+    activate api
+    api ->> api: 5. JWT署名は有効であるか?
+    api ->> api: 6. JWTの有効期限はどうか?
+    api ->> api: 7. ユーザ認証はどうか?
```
Comment (atoato88) [inline content/ja/docs/reference/access-authn-authz/authentication.md:218]: Original word is ["authorized"](https://gist.github.com/inductor/18a9278caa84436a6ee82cc80f12c38f#file-en-diff-L73), so you sould use "認可" not "認証"?

File: content/ja/docs/reference/access-authn-authz/authentication.md [hunk 5]
```diff
@@ -195,7 +195,35 @@ Secretは常にbase64でエンコードされるため、これらの値もbase6
...
+    deactivate idp
+    activate user
+    user ->> kube: 3. Kubectlを使用する場合、<br>--tokenフラグでid_tokenを使用するか、<br>または.kube/configにトークンを追加する
+    deactivate user
+    activate kube
+    kube ->> api: 4. 認可: Bearer...
+    deactivate kube
+    activate api
+    api ->> api: 5. JWT署名は有効であるか?
+    api ->> api: 6. JWTの有効期限はどうか?
```
Comment (atoato88) [inline content/ja/docs/reference/access-authn-authz/authentication.md:217]: missing `(iat+exp)` ?

File: content/ja/docs/reference/access-authn-authz/authentication.md (hunk 1 above)
Comment (ystkfujii) [reply content/ja/docs/reference/access-authn-authz/authentication.md]: Fixed !

File: content/ja/docs/reference/access-authn-authz/authentication.md (hunk 3 above)
Comment (ystkfujii) [reply content/ja/docs/reference/access-authn-authz/authentication.md:530]: Fixed !

File: content/ja/docs/reference/access-authn-authz/authentication.md (hunk 4 above)
Comment (ystkfujii) [reply content/ja/docs/reference/access-authn-authz/authentication.md]: Fixed !

File: content/ja/docs/reference/access-authn-authz/authentication.md (hunk 5 above)
Comment (ystkfujii) [reply content/ja/docs/reference/access-authn-authz/authentication.md]: Fixed !

**Comment by atoato88** [comment]
/lgtm
Thank you for quick fixing!

**Comment by atoato88** [comment]
[As bot said](https://github.com/kubernetes/website/pull/28376#issuecomment-859735645), you can assign reviewers for approving this PR.
My power is LGTM only. :sweat_smile:

File: content/ja/docs/reference/access-authn-authz/authentication.md [hunk 6]
```diff
@@ -195,7 +195,35 @@ Secretは常にbase64でエンコードされるため、これらの値もbase6
...
+    idp -->> user: 2. access_tokenとid_token、<br>refresh_tokenを提供
+    deactivate idp
+    activate user
+    user ->> kube: 3. Kubectlを使用する場合、<br>--tokenフラグでid_tokenを使用するか、<br>または.kube/configにトークンを追加する
+    deactivate user
+    activate kube
+    kube ->> api: 4. 認可: Bearer...
+    deactivate kube
+    activate api
+    api ->> api: 5. JWT署名は有効であるか?
```
Comment (inductor) [inline content/ja/docs/reference/access-authn-authz/authentication.md:216]:
```suggestion
    api ->> api: 5. JWT署名は有効か?
```

File: content/ja/docs/reference/access-authn-authz/authentication.md [hunk 7]
```diff
@@ -195,7 +195,35 @@ Secretは常にbase64でエンコードされるため、これらの値もbase6
...
+    api ->> api: 6. JWTの有効期限はどうか?(iat+exp)
```
Comment (inductor) [inline content/ja/docs/reference/access-authn-authz/authentication.md:217]:
```suggestion
    api ->> api: 6. JWTは期限切れでないか?(iat+exp)
```

File: content/ja/docs/reference/access-authn-authz/authentication.md [hunk 8]
```diff
@@ -195,7 +195,35 @@ Secretは常にbase64でエンコードされるため、これらの値もbase6
...
+    api ->> api: 7. ユーザ認可はどうか?
```
Comment (inductor) [inline content/ja/docs/reference/access-authn-authz/authentication.md:218]:
```suggestion
    api ->> api: 7. ユーザは認可済みか?
```

**Comment by inductor** [comment]
もとの英語的にこのほうがよさそう

**Comment by ystkfujii** [comment]
/assign @nasa9084
/assign @inductor
If this is OK, please approve.

**Pull request merged**